cvmfs/quota_posix.cc

```diff
@@ -1922,3 +1929,19 @@ void PosixQuotaManager::UnregisterBackChannel(
     ClosePipe(back_channel);
   }
 }
+
+void PosixQuotaManager::ManagedReadHalfPipe(int fd, void *buf, size_t nbyte, pid_t cachemgr_pid) {
+  unsigned timeout_ms = 100;
+  do {
+  ReadHalfPipe(fd, buf, nbyte, timeout_ms);
```

Comment:
```suggestion
    ReadHalfPipe(fd, buf, nbyte, timeout_ms);
```
Also, we need to return if `ReadHalfPipe` was successful.